provider/ansible/selflink.rb

```diff
@@ -38,7 +38,7 @@ def selflink_functions(object)
         virtuals = virtual_selflink_rrefs(object).map(&:resource_ref)
                                                  .uniq
         virtuals.map do |virt|
-          if virt = virtuals.last
+          if virt == virtuals.last
```

Comment:
Using .last inside a .map is fishy. You're into a collection loop and relying on the order. Let's not do that. If you need to do that either slice the collection or use an explicit for loop.